compile.sh

```diff
@@ -33,96 +25,190 @@ missingDepsExit ()
-executables. If you prefer ready-to-use binaries or if you want
-more info, please visit:  http://cuberite.org/"
+greeting ()
+{
+	# Do not greet when running with arguments.
+	if [ "$BRANCH" = "" -a "$BUILDTYPE" = "" ]; then
+		# Do we already have a repo?
+		if [ -d .git ] && [ -f easyinstall.sh ] && [ -f src/BlockArea.cpp ]; then # A good enough indicator that we're in the Cuberite git repo.
+			cd ../
+			echo "Cuberite repository detected. This should make the process faster, especially if you compiled before."
+			echo
+		fi
+
+		# Echo: Greetings.
+		echo "Hello, this script will download and compile Cuberite."
+		echo "On subsequent runs, it will update Cuberite."
+		echo "The compilation and download will occur in the current directory."
+		echo "If you're updating, you should run: <Path to Cuberite>/compile.sh"
+		echo "Compiling from source takes time, but it usually generates faster"
+		echo "executables. If you prefer ready-to-use binaries or if you want"
+		echo "more info, please visit:  http://cuberite.org/"
+	fi
+}
 
-### Dependency checks start. ###
-MISSING_PACKAGES=""
+dependencyCheck ()
+{
+	MISSING_PACKAGES=""
 
-# Most distros have the following default package and executable names.
-GCC_EXE_NAME="g++"
-CLANG_EXE_NAME="clang"
-COMPILER_PACKAGE_NAME="gcc g++"
+	# Most distros have the following default package and executable names.
+	GCC_EXE_NAME="g++"
+	CLANG_EXE_NAME="clang"
+	COMPILER_PACKAGE_NAME="gcc g++"
 
-# Left side: Executable Name, Right side: Package Name. Note that this is TAB delimited. Spaces will not work.
-PROGRAMS='git	git
+	# Left side: Executable Name, Right side: Package Name. Note that this is TAB delimited. Spaces will not work.
+	PROGRAMS='git	git
 make	make
 cmake	cmake'
 
-# If any OS deviates from the defaults, detect the OS here, and change PROGRAMS, COMPILER_PACKAGE_NAME, etc. as needed.
+	# If any OS deviates from the defaults, detect the OS here, and change PROGRAMS, COMPILER_PACKAGE_NAME, etc. as needed.
 
-# Fedora, CentOS, RHEL, Mageia, openSUSE, Mandriva
-if (rpm --help > /dev/null 2> /dev/null); then
-	COMPILER_PACKAGE_NAME="gcc-c++"
-fi
+	# Fedora, CentOS, RHEL, Mageia, openSUSE, Mandriva
+	if (rpm --help > /dev/null 2> /dev/null); then
+		COMPILER_PACKAGE_NAME="gcc-c++"
+	fi
 
-# Compiler check.
-GCC_EXISTS=0
-CLANG_EXISTS=0
-$GCC_EXE_NAME --help > /dev/null 2> /dev/null && GCC_EXISTS=1
-$CLANG_EXE_NAME --help > /dev/null 2> /dev/null && CLANG_EXISTS=1
-if [ "$GCC_EXISTS" -eq 0 ] && [ "$CLANG_EXISTS" -eq 0 ]; then
-	MISSING_PACKAGES=" $COMPILER_PACKAGE_NAME"
-fi
+	# Compiler check.
+	GCC_EXISTS=0
+	CLANG_EXISTS=0
+	$GCC_EXE_NAME --help > /dev/null 2> /dev/null && GCC_EXISTS=1
+	$CLANG_EXE_NAME --help > /dev/null 2> /dev/null && CLANG_EXISTS=1
+	if [ "$GCC_EXISTS" -eq 0 ] && [ "$CLANG_EXISTS" -eq 0 ]; then
+		MISSING_PACKAGES=" $COMPILER_PACKAGE_NAME"
+	fi
 
-# Depdendency check.
-checkPackages ()
+	# Depdendency check.
+	checkPackages ()
```

Comment:
Should be POSIX because such functions are basically just the definition of a 'comand' (like a subscript with a subshell) 

Please correct me if it's not POSIX